src/pages/ProgramContentPage/ProgramContentBlock.tsx

```diff
@@ -372,32 +391,49 @@ const useHasProgramContentPermission: (
   programContentId: string,
 ) => {
   hasProgramContentPermission: boolean
+  isLoading: boolean
 } = (programId, programContentId) => {
   const { currentMemberId, authToken } = useAuth()
-  const [data, setData] = useState<{ programContentId: string } | {}>({})
+  const [hasProgramContentPermission, setHasProgramContentPermission] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+  const isUnmounted = useRef(false)
 
-  const fetch = useCallback(async () => {
-    if (currentMemberId && programId && programContentId) {
-      const route = `/programs/${programId}/content/${programContentId}`
-      try {
-        const { data } = await axios.get(`${process.env.REACT_APP_LODESTAR_SERVER_ENDPOINT}${route}`, {
-          params: { memberId: currentMemberId },
-          headers: { authorization: `Bearer ${authToken}` },
-        })
+  const fetch = useCallback(
+    async (source: CancelTokenSource) => {
+      if (currentMemberId && programId && programContentId) {
+        const route = `/programs/${programId}/content/${programContentId}`
+        try {
+          const { data } = await axios.get(`${process.env.REACT_APP_LODESTAR_SERVER_ENDPOINT}${route}`, {
+            params: { memberId: currentMemberId },
+            headers: { authorization: `Bearer ${authToken}` },
+            cancelToken: source.token,
+          })
 
-        setData(data)
-      } catch (err) {
-        console.log(err)
+          if (!isUnmounted.current) {
+            setHasProgramContentPermission(Object.keys(data).length > 0)
+            setIsLoading(false)
+          }
+        } catch (err) {
+          !axios.isCancel(err) && console.log(err)
+        }
       }
-    }
-  }, [currentMemberId, programContentId, programId])
+    },
+    [currentMemberId, programContentId, programId],
+  )
```

Comment:
Is the 'isUnmounted' parameter necessary?
```
          await axios.get(`${process.env.REACT_APP_LODESTAR_SERVER_ENDPOINT}${route}`, {
            params: { memberId: currentMemberId },
            headers: { authorization: `Bearer ${authToken}` },
            cancelToken: source.token,
          }).then(({ data }) => {
            setHasProgramContentPermission(Object.keys(data).length > 0)
            setIsLoading(false)
          })
```

Reply:
Although there is a mechanism for canceling requests, this judgment is for even more secure prevention, ensuring that other actions are not executed after the component has already been destroyed.